builtin/providers/aws/resource_aws_security_group.go

```diff
@@ -407,6 +411,23 @@ func resourceAwsSecurityGroupUpdateRules(
 		ns := n.(*schema.Set)
 
 		remove := expandIPPerms(group, os.Difference(ns).List())
+
+		if len(remove) == 0 && len(os.List()) == 0 && ruleset == "egress" {
```

Comment:
Probably technically redundant, but I feel like it doesn't hurt to be extra explicit/conservative.